src/main/java/com/smartsheet/api/internal/ReportResourcesImpl.java

```diff
@@ -0,0 +1,251 @@
+        {
+            throw new IllegalArgumentException("Page Size must be between 1 and 500.");
+        }
+    }
+
+    /**
+     * List all reports.
+     *
+     * It mirrors to the following Smartsheet REST API method: GET /reports
+     *
+     * Exceptions:
+     *   - InvalidRequestException : if there is any problem with the REST API request
+     *   - AuthorizationException : if there is any problem with the REST API authorization(access token)
+     *   - ServiceUnavailableException : if the REST API service is not available (possibly due to rate limiting)
+     *   - SmartsheetRestException : if there is any other REST API related error occurred during the operation
+     *   - SmartsheetException : if there is any other error occurred during the operation
+     *
+     * @return all sheets (note that empty list will be returned if there is none)
+     * @throws SmartsheetException the smartsheet exception
+     */
+    public DataWrapper<Report> listReports(PaginationParameters parameters) throws SmartsheetException {
+        String path= "reports";
+        if (parameters != null) {
+            path += parameters.toQueryString();
+            }
+        return this.listResourcesWithWrapper(path, Report.class);
+    }
+
+    /**
+     * Get a Report as an Excel file.
+     *
+     * It mirrors to the following Smartsheet REST API method: GET /reports/{id} with "application/vnd.ms-excel" Accept
+     * HTTP header
+     *
+     * Exceptions:
+     *   IllegalArgumentException : if outputStream is null
+     *   InvalidRequestException : if there is any problem with the REST API request
+     *   AuthorizationException : if there is any problem with the REST API authorization(access token)
+     *   ResourceNotFoundException : if the resource can not be found
+     *   ServiceUnavailableException : if the REST API service is not available (possibly due to rate limiting)
+     *   SmartsheetRestException : if there is any other REST API related error occurred during the operation
+     *   SmartsheetException : if there is any other error occurred during the operation
+     *
+     * @param id the id
+     * @param includes the optional objects to include in response
+     * @param pageSize Number of rows per page
+     * @param page page number to return
+     * @param outputStream the OutputStream to which the Excel file will be written
+     * @return the sheet as csv
+     * @throws SmartsheetException the smartsheet exception
+     */
+    public void getReportAsExcel(long id, EnumSet<ReportInclusion> includes, Integer pageSize, Integer page, OutputStream outputStream) throws SmartsheetException {
+        this.checkParameters(pageSize, page);
+
+        String path = "/reports";
+        HashMap<String, String> parameters = new HashMap<String, String>();
+        parameters.put("include", QueryUtil.generateCommaSeparatedList(includes));
+
+        if (pageSize != null) {
```

Comment:
no need for null check, just pass page & pageSize directly without calling toString() on them.